lib/pragma-grammar.ym

```diff
@@ -123,7 +123,7 @@ module_stmt
 requires_stmt
 	: KW_REQUIRES string
           {
-            if (!plugin_load_module(&configuration->plugin_context, $2, NULL))
+            if (!plugin_is_module_available(&configuration->plugin_context, $2))
```

Comment:
Now, `KW_REQUIRES` depends on the initial discovery of candidate modules.

This currently appears on the lexer level, candidates are loaded after the first non-pragma token.
Are we sure this fix is correct? I mean, `@include` and `@requires` are pragma tokens.

Reply:
I would consider that plugin discovery is pretty much turned on everywhere, so we could rely on that. However you have a point that `@requires` might come in the first couple of lines, in which case plugin discovery has not run yet. I don't see this as a very important usecase, but I agree this should be handled with a bit more care.

Reply:
> I would consider that plugin discovery is pretty much turned on everywhere, so we could rely on that.

I agree.

> However you have a point that `@requires` might come in the first couple of lines, in which case plugin discovery has not run yet.

If the first included SCL file has any `@requires` pragma, that SCL will be skipped incorrectly.
This can be fragile, and since we use `@include 'scl/*/*.conf'`, the bug depends on the file traversal order.

We should fix that before switching to the new and better `plugin_is_module_available()` check.

Reply:
I have now resolved this comment in the latest rebase.